test/requests/code_lens_expectations_test.rb

```diff
@@ -130,11 +130,13 @@ def test_code_lens_addons
       class Test < Minitest::Test; end
     RUBY
 
-    test_addon(:create_code_lens_addon, source: source) do |server|
+    create_code_lens_addon
```

Comment:
I know it's a bit weird to rely on `with_server`'s ensure block to clear the anonymous addon implicitly. But IMO the weirdness comes these tests' need of 1-time addons. It won't be the case in real addons' tests.